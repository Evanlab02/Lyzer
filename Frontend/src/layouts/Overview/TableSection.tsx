import Card from "../../components/Card";
import { GridItem } from "../../components/Grid";

export default function TableSection() {
	return (
		<>
			<GridItem xs={12} sm={12} md={6} lg={4} xl={3} xxl={3}>
				<Card>
					<h1>Overview Section</h1>
				</Card>
			</GridItem>
			<GridItem xs={12} sm={12} md={6} lg={4} xl={4} xxl={4}>
				<Card>
					<h1>Constructor Standings</h1>
				</Card>
			</GridItem>
			<GridItem xs={12} sm={12} md={6} lg={4} xl={5} xxl={5}>
				<Card>
					<h1>Driver Standings</h1>
				</Card>
			</GridItem>
		</>
	);
}