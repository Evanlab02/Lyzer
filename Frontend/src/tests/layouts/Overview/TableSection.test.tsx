import { expect, it } from "vitest";
import { render } from "@testing-library/react";
import TableSection from "../../../layouts/Overview/TableSection";

it("TableSection renders loading state correctly", async () => {
	const { findByTestId } = render(
		<div data-testid="lyzer-table-section">
			<TableSection seasonProgress={undefined} />
		</div>
	);

	const tableSection = await findByTestId("lyzer-table-section");
	expect(tableSection).toMatchSnapshot();
});

it("TableSection renders correctly", async () => {
	const { findByTestId } = render(
		<div data-testid="lyzer-table-section">
			<TableSection 
				seasonProgress={{
					previousRaceWinner: "Max Verstappen",
					previousGrandPrix: "Qatar",
					seasonProgress: 12,
					seasonTotalRaces: 24
				}}
			/>
		</div>
	);

	const tableSection = await findByTestId("lyzer-table-section");
	expect(tableSection).toMatchSnapshot();
});
