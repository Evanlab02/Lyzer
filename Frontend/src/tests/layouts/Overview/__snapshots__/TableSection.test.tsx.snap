// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableSection renders correctly 1`] = `
<div
  data-testid="lyzer-table-section"
>
  <div
    class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-12 flex-grid-col-lg-12 flex-grid-col-xl-3 flex-grid-col-xxl-3"
  >
    <div
      class="lyzer-card"
    >
      <h3
        class="lyzer-card-body"
      >
        <div
          class="lyzer-card-section"
        >
          <h3
            class="lyzer-card-section-title"
          >
            Max Verstappen
          </h3>
          <p
            class="lyzer-card-section-subtitle"
          >
            Previous Race Winner
          </p>
          <hr
            class="lyzer-card-section-divider"
          />
        </div>
        <div
          class="lyzer-card-section"
        >
          <h3
            class="lyzer-card-section-title"
          >
            Qatar
          </h3>
          <p
            class="lyzer-card-section-subtitle"
          >
            Previous Grand Prix
          </p>
          <hr
            class="lyzer-card-section-divider"
          />
        </div>
        <div
          class="lyzer-card-section"
        >
          <h3
            class="lyzer-card-section-title"
          >
            12 of 24 races
          </h3>
          <p
            class="lyzer-card-section-subtitle"
          >
            Season Progress
          </p>
          <hr
            class="lyzer-card-section-divider"
          />
        </div>
      </h3>
    </div>
  </div>
  <div
    class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-12 flex-grid-col-lg-12 flex-grid-col-xl-9 flex-grid-col-xxl-9"
  >
    <div
      class="lyzer-card"
    >
      <span
        class="lyzer-card-heading"
      >
        Constructor Standings
      </span>
    </div>
  </div>
  <div
    class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-12 flex-grid-col-lg-12 flex-grid-col-xl-12 flex-grid-col-xxl-12"
  >
    <div
      class="lyzer-card"
    >
      <span
        class="lyzer-card-heading"
      >
        Driver Standings
      </span>
    </div>
  </div>
</div>
`;

exports[`TableSection renders loading state correctly 1`] = `
<div
  data-testid="lyzer-table-section"
>
  <div
    class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-12 flex-grid-col-lg-12 flex-grid-col-xl-3 flex-grid-col-xxl-3"
  >
    <div
      class="lyzer-card"
    >
      <h3
        class="lyzer-card-body"
      >
        <div
          class="lyzer-card-section"
        >
          <h3
            class="lyzer-card-section-title"
          >
            Loading...
          </h3>
          <p
            class="lyzer-card-section-subtitle"
          >
            Previous Race Winner
          </p>
          <hr
            class="lyzer-card-section-divider"
          />
        </div>
        <div
          class="lyzer-card-section"
        >
          <h3
            class="lyzer-card-section-title"
          >
            Loading...
          </h3>
          <p
            class="lyzer-card-section-subtitle"
          >
            Previous Grand Prix
          </p>
          <hr
            class="lyzer-card-section-divider"
          />
        </div>
        <div
          class="lyzer-card-section"
        >
          <h3
            class="lyzer-card-section-title"
          >
            Loading...
          </h3>
          <p
            class="lyzer-card-section-subtitle"
          >
            Season Progress
          </p>
          <hr
            class="lyzer-card-section-divider"
          />
        </div>
      </h3>
    </div>
  </div>
  <div
    class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-12 flex-grid-col-lg-12 flex-grid-col-xl-9 flex-grid-col-xxl-9"
  >
    <div
      class="lyzer-card"
    >
      <span
        class="lyzer-card-heading"
      >
        Constructor Standings
      </span>
    </div>
  </div>
  <div
    class="flex-grid-item flex-grid-col-xs-12 flex-grid-col-sm-12 flex-grid-col-md-12 flex-grid-col-lg-12 flex-grid-col-xl-12 flex-grid-col-xxl-12"
  >
    <div
      class="lyzer-card"
    >
      <span
        class="lyzer-card-heading"
      >
        Driver Standings
      </span>
    </div>
  </div>
</div>
`;
